app/code/Dev/ProductComments/Block/Widget/Samplewidget.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace Dev\ProductComments\Block\Widget;
+
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\View\Element\Template;
+use Magento\Widget\Block\BlockInterface;
+
+class Samplewidget extends Template implements BlockInterface
+{
+
+    protected $_template = "widget/samplewidget.phtml";
```

Comment:
same here, I don't like the name of the .phtml file. For phtml files, you should not use cames cases but divide with "-". In your case, it could be sample-widget.phtml. But better comments-widget.phtml